app/src/org/commcare/models/database/DbUtil.java

```diff
@@ -8,107 +8,14 @@
 import net.sqlcipher.database.SQLiteDatabase;
 import net.sqlcipher.database.SQLiteDatabaseHook;
 
-import org.commcare.models.AndroidPrototypeFactory;
 import org.commcare.modern.database.DatabaseHelper;
-import org.javarosa.core.util.PrefixTree;
-import org.javarosa.core.util.externalizable.Externalizable;
-import org.javarosa.core.util.externalizable.PrototypeFactory;
 
 import java.io.File;
-import java.io.IOException;
-import java.lang.reflect.Constructor;
-import java.util.ArrayList;
-import java.util.Enumeration;
-import java.util.List;
-
-import dalvik.system.DexFile;
 
 public class DbUtil {
     private static final String TAG = DbUtil.class.getSimpleName();
     public final static String orphanFileTableName = "OrphanedFiles";
 
-    private static PrototypeFactory factory;
-    private static final String[] packageNames = new String[]{"org.javarosa", "org.commcare", "org.odk.collect"};
-
-    public static void setDBUtilsPrototypeFactory(PrototypeFactory factory) {
-        DbUtil.factory = factory;
-    }
-
-    /**
-     * Basically this is our PrototypeManager for Android
-     */
-    public static PrototypeFactory getPrototypeFactory(Context c) {
-        if (factory != null) {
-            return factory;
-        }
-
-        PrefixTree tree = new PrefixTree();
-
-        try {
-            List<String> classes = getClasses(c);
-            for (String cl : classes) {
-                tree.addString(cl);
-            }
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-
-        factory = new AndroidPrototypeFactory(tree);
-        return factory;
-    }
-
-    /**
-     * Scans all classes accessible from the context class loader which belong to the given package and subpackages.
-     */
-    private static List<String> getClasses(Context c)
-            throws IOException {
-        ArrayList<String> classNames = new ArrayList<>();
-
-        String zpath = c.getApplicationInfo().sourceDir;
-
-
-        if (zpath == null) {
-            zpath = "/data/app/org.commcare.android.apk";
-        }
-
-        DexFile df = new DexFile(new File(zpath));
-        for (Enumeration<String> en = df.entries(); en.hasMoreElements(); ) {
-            String cn = en.nextElement();
-            loadClass(cn, classNames);
-        }
-
-        return classNames;
-    }
-
-    public static void loadClass(String cn, List<String> classNames) {
-        try {
-            for (String packageName : packageNames) {
-                if (cn.startsWith(packageName) && !cn.contains(".test.") && !cn.contains("readystatesoftware")) {
```

Comment:
removed `&& !cn.contains(".test.") && !cn.contains("readystatesoftware")` from the moved code.